src/lib/x509/certstor.h

```diff
@@ -31,6 +31,15 @@ class BOTAN_DLL Certificate_Store
          find_cert(const X509_DN& subject_dn, const std::vector<byte>& key_id) const = 0;
 
       /**
+      * Find a certificate by searching for one with a matching SHA-1 hash of
+      * public key. Used for OCSP.
+      * @param key_hash SHA-1 hash of the subject's public key
+      * @return a matching certificate or nullptr otherwise
+      */
+      virtual std::shared_ptr<const X509_Certificate>
+         find_cert_by_pubkey_sha1(const std::vector<byte>& key_hash) const = 0;
```

Comment:
It definitely is not pretty. We could make SHA-1 optional and have requests be issued with SHA-256 (causing some responders to reject our requests) and ignore/fail any replies that are identified with a key hash. That's a substantial hit to functionality, but maybe some closed systems who control their OCSP responders would still find it usable.

Alternately we can split out OCSP from X509, but I would really rather not because it would complicate the API (having to have 2 different functions where there is currently 1: one consuming OCSP the other not).